unison-share-api/src/Unison/Server/CodebaseServer.hs

```diff
@@ -146,16 +147,72 @@ data BaseUrl = BaseUrl
     urlPort :: Port
   }
 
-data BaseUrlPath = UI | Api
+data NameOrHashRef = Name [Text] | Hash Text
+
+data DefinitionReference
+  = TermReference NameOrHashRef
+  | TypeReference NameOrHashRef
+  | AbilityConstructorReference NameOrHashRef
+  | DataConstructorReference NameOrHashRef
+
+data BaseUrlPath
```

Comment:
Should this maybe accept the domain types, e.g. `Path` and `HashQualified Name` instead of Texts, then have the URI builder convert them at the last second?

Reply:
Yeah that sounds like a good idea!